torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -749,4 +749,78 @@ xla::XlaOp BuildMaskedScatter(xla::XlaOp input, xla::XlaOp mask,
       scatter_dnums);
 }
 
+std::vector<xla::XlaOp> BuildAmpForeachNonFiniteCheckAndUnscale(
+    const std::vector<xla::XlaOp>& inputs, const xla::XlaOp& found_inf_float,
+    const xla::XlaOp& inv_scale) {
+  const xla::PrimitiveType origin_type =
+      XlaHelpers::ShapeOfXlaOp(found_inf_float).element_type();
+  std::vector<xla::XlaOp> found_infs;
+  xla::XlaOp one = xla::One(inputs[0].builder(), xla::PrimitiveType::PRED);
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    xla::XlaOp all_finite =
+        xla::ReduceAll(xla::IsFinite(inputs[i]), one,
+                       xla::CreateScalarAndComputation(xla::PrimitiveType::PRED,
+                                                       inputs[i].builder()));
+    found_infs.push_back(xla::Not(all_finite));
+  }
+  xla::XlaOp found_inf =
+      xla::ConvertElementType(found_inf_float, xla::PrimitiveType::PRED);
+  for (size_t i = 0; i < found_infs.size(); ++i) {
+    found_inf = xla::Or(found_inf, found_infs[i]);
+  }
```

Comment:
nit, can we just do 
```
found_inf = xla::Or(found_inf, xla::Not(all_finite));
```
in the first for loop? it doesn't seems like you need  `found_infs` in later codes

Reply:
Done.